strapi/strapi
# Fix RouterConfig type

### What does it do?

Update the `RouterConfig` type, set `only` as optional.

### Why is it needed?

`only` isn't really required. In `packages/core/strapi/lib/factories.js`, `only` is checked before being used (line 69):

```
const selectedRoutes = pipe(
      (routes) => (except ? omit(except, routes) : routes),
      (routes) => (only ? pick(only, routes) : routes)
)(defaultRoutes);
```

### How to test it?

[Configure a core router](https://docs.strapi.io/developer-docs/latest/development/backend-customization/routes.html#configuring-core-routers), do not set the `only` parameter. TypeScript should not complain any more, and Strapi will start without errors.

### Related issue(s)/PR(s)

None AFAIK.
